### PR TITLE
fix: do not log error for not existing folder during resource scan

### LINKED ISSUE #18855: PiT 24.4: regression mvn vaadin:clean-frontend and mvn vaadin:dance dont work
### Description of the bug

`vaadin:clean-frontend` and `vaadin:dance` do not work, causing the following error:

```
could not create Vfs.Dir from url. ignoring the exception and continuing
org.reflections.ReflectionsException: could not create Vfs.Dir from url, no matching UrlType was found [file:/Users/manolo/Github/platform/platform-in-test-script/tmp/skeleton-starter-flow-spring/target/classes]
either use fromURL(final URL url, final List<UrlType> urlTypes) or use the static setDefaultURLTypes(final List<UrlType> urlTypes) or addDefaultURLTypes(UrlType urlType) with your specialized UrlType.
    at org.reflections.vfs.Vfs.fromURL (Vfs.java:113)
    at org.reflections.vfs.Vfs.fromURL (Vfs.java:95)
    at org.reflections.Reflections.lambda$scan$2 (Reflections.java:176)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept (ForEachOps.java:183)
    at java.util.HashMap$KeySpliterator.forEachRemaining (HashMap.java:1707)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:509)
    at java.util.stream.ForEachOps$ForEachTask.compute (ForEachOps.java:290)
    at java.util.concurrent.CountedCompleter.exec (CountedCompleter.java:754)
    at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:373)
    at java.util.concurrent.ForkJoinTask.invoke (ForkJoinTask.java:686)
    at java.util.stream.ForEachOps$ForEachOp.evaluateParallel (ForEachOps.java:159)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel (ForEachOps.java:173)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:233)
    at java.util.stream.ReferencePipeline.forEach (ReferencePipeline.java:596)
    at java.util.stream.ReferencePipeline$Head.forEach (ReferencePipeline.java:765)
    at org.reflections.Reflections.scan (Reflections.java:173)
    at org.reflections.Reflections.<init> (Reflections.java:126)
    at com.vaadin.flow.server.scanner.ReflectionsClassFinder$LoggingReflections.<init> (ReflectionsClassFinder.java:120)
    at com.vaadin.flow.server.scanner.ReflectionsClassFinder.<init> (ReflectionsClassFinder.java:66)
    at com.vaadin.flow.plugin.base.BuildFrontendUtil.getClassFinder (BuildFrontendUtil.java:110)
    at com.vaadin.flow.plugin.maven.FlowModeAbstractMojo.getClassFinder (FlowModeAbstractMojo.java:318)
    at com.vaadin.flow.plugin.base.CleanFrontendUtil.runCleaning (CleanFrontendUtil.java:66)
    at com.vaadin.flow.plugin.maven.CleanFrontendMojo.execute (CleanFrontendMojo.java:45)
```

### Expected behavior

it works

### Minimal reproducible example

```
git clone https://github.com/vaadin/skeleton-starter-flow-spring.git
cd skeleton-starter-flow-spring
## Change Maven property vaadin.version from 24.3.5 -> 24.4-SNAPSHOT
perl -0777 -pi -e 's|(<vaadin.version>)([^\s]+)(</vaadin.version>)|${1}24.4-SNAPSHOT${3}|g' pom.xml
## Add prereleases repos
perl -pi -e 's|(\s*)(<repositories>)|$1$2\n$1$1<repository><id>v</id><url>https://maven.vaadin.com/vaadin-prereleases</url></repository>|' pom.xml
perl -pi -e 's|(\s*)(</properties>)|$1$2\n$1<pluginRepositories><pluginRepository><id>v</id><url>https://maven.vaadin.com/vaadin-prereleases</url></pluginRepository></pluginRepositories>|' pom.xml
## run the project
mvn
```

### Versions

- Vaadin / Flow version: 24.4.0.alpha7
- Mac OS, Linux, Windows

### PR DESCRIPTION
**Pull request by mcollovati**
## Description

Prevents a warning message to be printed when the class finder scanner is not able to handle an URL, for example because it points to a not existing directory.
A message is anyway printed at debug level.

Fixes #18855

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: flow-plugins/flow-plugin-base/src/main/java/com/vaadin/flow/server/scanner/ReflectionsClassFinder.java [hunk 1]
```diff
@@ -164,4 +173,43 @@ public Class<?> forClass(String typeName, ClassLoader... loaders) {
...
         }
     }
+
+    private static final Vfs.UrlType IGNORE_NOT_HANDLED_FILES = new Vfs.UrlType() {
+
+        public boolean matches(URL url) {
+            if (!"file".equals(url.getProtocol())) {
+                return false;
+            }
+            for (Vfs.DefaultUrlTypes urlTypes : EnumSet.of(
```
[the remark this answers is not in the document]
Comment (mcollovati) [reply flow-plugins/flow-plugin-base/src/main/java/com/vaadin/flow/server/scanner/ReflectionsClassFinder.java:183]: Dir and jar are the only types handling file protocol.
If they are not able to handle the URL, the warning message will be printed. So, in that case we provide this fake implementation

[the remark this answers is not in the document]
Comment (mcollovati) [reply flow-plugins/flow-plugin-base/src/main/java/com/vaadin/flow/server/scanner/ReflectionsClassFinder.java:183]: Actually, this instance should be the first that is checked

[the remark this answers is not in the document]
Comment (tltv) [reply flow-plugins/flow-plugin-base/src/main/java/com/vaadin/flow/server/scanner/ReflectionsClassFinder.java:183]: Aha, indeed it's added to 0-index. That explains the need for the "double-checking". But how about moving it to end and removing the loop. It should work like that too with a bit less code.

[the remark this answers is not in the document]
Comment (mcollovati) [reply flow-plugins/flow-plugin-base/src/main/java/com/vaadin/flow/server/scanner/ReflectionsClassFinder.java:183]: Sounds a good idea.

[the remark this answers is not in the document]
Comment (mcollovati) [reply flow-plugins/flow-plugin-base/src/main/java/com/vaadin/flow/server/scanner/ReflectionsClassFinder.java]: done

**Review by tltv — APPROVED** [review]

**Pull request merged**